launcher/src/main/java/org/apache/spark/launcher/SparkAppHandle.java

```diff
@@ -43,6 +43,8 @@
     RUNNING(false),
     /** The application finished with a successful status. */
     FINISHED(true),
+    /** The Spark Submit JVM got killed with a unknown status. */
+    FINISHED_UNKNOWN(true),
```

Comment:
This should be declared after existing constants, since declaring it here will change the `ordinal()` value of existing constants, which is a binary compatibility issue. A better name would be good also; maybe "LOST" or "DISCONNECTED"?